modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/components/Field/Field.es.js

```diff
@@ -111,39 +119,71 @@ const mountStruct = (event, field, value) => {
 	return new FieldEventStruct(event, field, value);
 };
 
-export const Field = ({field, onBlur, onChange, onFocus, ...otherProps}) => {
-	const {fieldTypes} = usePage();
-	const [hasError, setHasError] = useState(false);
-	const loadField = useLazy();
-
+const FieldLazy = ({
+	field,
+	fieldTypes,
+	onBlur,
+	onChange,
+	onFocus,
+	...otherProps
+}) => {
 	const focusDurationRef = useRef({end: null, start: null});
 
+	const ComponentLazy = useLazy()(getModule(fieldTypes, field.type));
+
+	return (
+		<ComponentLazy
+			onBlur={(event) => {
+				focusDurationRef.current.end = new Date();
+				onBlur(mountStruct(event, field), focusDurationRef.current);
+			}}
+			onChange={(event, value) =>
+				onChange(mountStruct(event, field, value))
+			}
+			onFocus={(event) => {
+				focusDurationRef.current.start = new Date();
+				onFocus(mountStruct(event, field));
+			}}
+			visible
+			{...field}
+			{...otherProps}
+		/>
+	);
+};
+
+export const Field = ({field, ...otherProps}) => {
+	const {fieldTypes} = usePage();
+	const [hasError, setHasError] = useState();
+
 	if (!fieldTypes) {
 		return <ClayLoadingIndicator />;
 	}
 
-	const fieldModule = getModule(fieldTypes, field.type);
-	const FieldLazy = loadField(fieldModule);
-
 	if (hasError) {
 		return (
 			<div className="ddm-field-renderer--error">
 				<p className="ddm-field-renderer--title">
-					Oops! An error happening.
```

Comment:
Hehe. I didn't even notice this one the first time through.

Reply:
🤪

Reply:
Just to make sure we're on the same page with the problems on this line:

1. That should use `Liferay.Language.get()`
2. Not a native English speaker here, but the sentence seems off. Maybe @wincent can suggest something better. I'm thinking: "There was an error trying to render this field."

I've briefly talked about that to Matu some time ago, but I wonder if we should push the error boundary upwards to the <form> level. The reason is that the field might be **required**, so it doesn't make sense for the user to fill up the other fields if he is never gonna be able to submit the form.

I've seen a bug was created for this recently: https://issues.liferay.com/browse/LPS-116637

Reply:
Only now I noticed the diff is showing me he removed this line.

![tenor-2811121](https://user-images.githubusercontent.com/156388/87057384-2a4f6180-c1dd-11ea-9202-5ba7b9280325.gif)

Reply:
haha hey @brunobasto, yeah I changed according to that our conversation previously, and also changed the behavior of the button of just appearing in case of a network failure. About moving the error to the entire Form: I'm not sure, I think that the best would be to check with UX designer or team to check it better.

Reply:
Agreed. That's up to the team to decide.